sparse/utils.py

```diff
@@ -258,13 +287,53 @@ def equivalent(x, y):
 
 
 def check_zero_fill_value(*args):
+    """
+    Checks if all the arguments have zero fill-values.
+
+    Parameters
+    ----------
+    args : Iterable[COO]
+
+    Examples
+    --------
+    >>> import sparse
+    >>> s1 = sparse.random((10,), density=0.5)
+    >>> s2 = sparse.random((10,), density=0.5, fill_value=0.5)
+    >>> check_zero_fill_value(s1)
+    >>> check_zero_fill_value(s2)
+    Traceback (most recent call last):
+        ...
+    ValueError: This operation requires zero fill values.
+    >>> check_zero_fill_value(s1, s2)
+    Traceback (most recent call last):
+        ...
+    ValueError: This operation requires zero fill values.
+    """
     for arg in args:
         if (hasattr(arg, 'fill_value') and
                 not equivalent(arg.fill_value, _zero_of_dtype(arg.dtype))):
             raise ValueError('This operation requires zero fill values.')
```

Comment:
It's a minor point, but it's nice to include offending values in all error messages, e.g., `arg.fill_value` in this case

Reply:
I'm not sure what best practice would be here. We wouldn't know what exact argument would produce this fill value, so showing it may be not be useful. What do you think?

Reply:
Maybe:
```python
'This operation requires zero fill values, but argument {} has fill value {}'.format(i, arg.fill_value)
```
where `i` comes from iterating over `args` with `enumerate()`.

Reply:
Done.